File: 0078_Subsets.cpp
class Solution {
public:
    vector<vector<int>> subsets(vector<int>& nums) {
        vector<vector<int>> vec;
        std::sort(nums.begin(), nums.end());
        nums.erase(std::unique(nums.begin(), nums.end()), nums.end());
        const int N = nums.size();
        for (int i = 0; i < pow(2, N); i++){ //
            string s = std::bitset<N>(i);
            while (s.size() < N) s = "0" + s;
            vector<int> v;
            for (int j = 0; j < nums.size(); v++){
                if(s[j] == 1) v.push_back(nums[j]);
            }
            vec.push_back(v);
            /*int j = i;
            while (j != 0){
                if (j & 1 == 1){
                    
                }
                j >> 1;
            }*/
        }
        
        return vec;
    }
};

File: 0078_Subsets.py
class Solution:
    def subsets(self, nums: List[int]) -> List[List[int]]:
        bits = 0
        out = []
        while bits < 2 ** len(nums):
            l = []
            str_bits = bin(bits).replace('0b', '').zfill(len(nums))
            for i,s in enumerate(str_bits):
                if s == '1':
                    l.append(nums[i])
            out.append(l)
            bits += 1
            
        out = list(map(list, set(map(tuple, out))))
        return out
